packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart

```diff
@@ -141,7 +146,18 @@ void main() {
           printOnFailure(buildResult.stderr.toString());
           expect(buildResult.exitCode, 0);
 
-          expect(outputPluginFrameworkBinary, exists);
+          // Plugins are built either as a static library (SwiftPM's default)
+          // or as a dynamic library (CocoaPods's default).
+          // If built as a dynamic library, the plugin will have a .framework.
+          // If built as static library, the plugin's symbols will be in the
+          // Runner binary.
+          final bool helloDynamic = outputPluginFrameworkBinary.existsSync();
+          final bool helloStatic = AppleTestUtils
+            .getExportedSymbols(outputRunnerBinary.path)
+            .any((String symbol) => symbol.contains('HelloPlugin') && symbol.contains('handle'));
+
+          expect(helloDynamic || helloStatic, isTrue);
+          expect(helloDynamic != helloStatic, isTrue);
```

Comment:
This assertion is meant to be a logical xor ("Either built dynamically or statically, but not both"). Technically, the `!=` is the only assertion that's needed here but I added the `||` assertion for readability.